stdlib/SparseArrays/src/higherorderfns.jl

```diff
@@ -719,23 +719,15 @@ function _broadcast_notzeropres!(f::Tf, fillvalue, C::SparseVecOrMat, A::SparseV
             Bk, stopBk = numcols(B) == 1 ? (colstartind(B, 1), colboundind(B, 1)) : (colstartind(B, j), colboundind(B, j))
             Bx = Bk < stopBk ? storedvals(B)[Bk] : zero(eltype(B))
             fzAvB = f(zero(eltype(A)), Bx)
-            if _iszero(fzAvB)
```

Comment:
Having briefly refreshed my memory of these methods, I believe that the `_iszero(fzAvB)` check (and similar `_iszero(fvAzB)` check in the preceding branch where `A` rather than `B` expands vertically) should instead be something to the effect of `fzAvB == fillvalue` in the methods for the not-zero-preserving-f case. These branches should stay around, however, as otherwise performance may degrade appreciably. Regrettably a nanosoldier suite for these methods remains a "one day" project. Best!